src/prpy/planning/vectorfield.py

```diff
@@ -357,7 +357,7 @@ def fn_callback(t, q):
         output_path.Insert(0, path.GetWaypoints(0, cached_index), cspec)
 
         # Add a segment for the feasible part of the last integration step.
-        output_path.Insert(output_path.GetNumWaypoints() - 1,
+        output_path.Insert(output_path.GetNumWaypoints(),
```

Comment:
Looking at the openrave implementation of ```GenericTrajectory```, ```Insert``` puts the element at the specified index:

https://github.com/rdiankov/openrave/blob/b8fa1bb1fad5a86d373cc5d286e925f694540ea4/src/libopenrave-core/generictrajectory.cpp#L103

Here we want to append to the end of the trajectory, so passing ```output_path.GetNumWaypoints()``` should be correct.